projects/plugins/boost/app/assets/src/js/pages/settings/elements/CriticalCssStatus.svelte

```diff
@@ -0,0 +1,91 @@
+<script>
+	/**
+	 * WordPress dependencies
+	 */
+	import { __, _n, sprintf } from '@wordpress/i18n';
+	import { createEventDispatcher } from 'svelte';
+
+	/**
+	 * Internal dependencies
+	 */
+	import {
+		criticalCssStatus,
+		failedProviderKeyCount,
+		cssModuleName,
+	} from '../../../stores/critical-css-status';
+	import RefreshIcon from '../../../svg/refresh.svg';
+	import TimeAgo from '../../../elements/TimeAgo.svelte';
+	import InfoIcon from '../../../svg/info.svg';
+	import TemplatedString from '../../../elements/TemplatedString.svelte';
+	import actionLinkTemplateVar from '../../../utils/action-link-template-var';
+	import routerHistory from '../../../utils/router-history.ts';
+	import CriticalCssShowStopperError from './CriticalCssShowStopperError.svelte';
+
+	const dispatch = createEventDispatcher();
+	const { navigate } = routerHistory;
+
+	// Show an error if in error state, or if a success has 0 results.
+	let showError = false;
+	$: showError =
+		$criticalCssStatus.status === 'error' ||
+		( $criticalCssStatus.status === 'success' && $criticalCssStatus.success_count === 0 );
+</script>
+
+{#if $cssModuleName === 'critical-css' && $criticalCssStatus.status === 'requesting'}
+	<slot name="progress" />
+{:else if showError}
+	<CriticalCssShowStopperError on:retry={() => dispatch( 'retryShowStopper' )} />
+{:else}
+	<div class="jb-critical-css__meta">
+		<div class="summary">
+			{#if $criticalCssStatus.success_count === 0}
+				<div class="generating">
```

Comment:
I think this is more complicated than it needs to be.

1. It uses a mix of approaches to figure out what to show; it accepts properties and slots which tell it what to show / how to behave, then it reaches out to figure out whether the user is using local or cloud css itself (i.e.: checking `$cssModuleName`). Using both approaches in one component close to each other is a sign of complex dependencies.
2. There are two separate if/then paths used when generating Critical CSS, depending on whether it's generating locally or remotely. Is there a reason not to combine them? It would be better if this UI could just as the CSS datastore "are you generating?", and if so, just show a "generating" slot that is populated by its parent component.

Some possible ideas to clean this up a bit:

- Don't make the progress bar a child of the `CriticalCssStatus` (via slots) -- just have it directly inside the `CriticalCssMeta` component when Local Critcial CSS is generating. That way, the logic which only belongs to Local Critical CSS is set aside in one of its dedicated components. 
- Maybe break `CriticalCssStatus` into a number of smaller components which can be reused directly in `CriticalCssMeta` and `CloudCssMeta`. For example, the "X files failed, Visit the advanced recommendations" information could be a small standalone component.
- Alternately, maybe make `CriticalCssStatus` accept a single "Status" text line which describes the current status - putting the onus onto the parent components to describe the status? You could always move logic which has to be repeated (e.g.: "X files generated") into a utility function.

I'm not fixed on either of these approaches, but I find the current layout a bit difficult to follow as I feel too many components need to be aware of each other's inner workings at the moment.

Reply:
Moving the progress bar directly to `CriticalCssMeta` makes perfect sense. That leaves all the common bits in the `CriticalCssStatus` without the use of `$cssModuleName`. I have updated this